php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+    ) {
+        $this->_lorisForm      = $lorisForm;
+        $this->_moveDirPath    = is_null($moveDirPath)
+            ? self::$DEFAULT_MOVE_DIR : $moveDirPath;
+        $this->_dirPermissions = $dirPermissions;
+        $this->_dirGroup       = $dirGroup;
+
+        $this->_errors = array();
+    }
+
+    /**
+     * Adds an error (any kind) to the set of current errors.
+     *
+     * @param string $key      name of the LorisForm field associated to the
+     *                         uploaded file.
+     * @param string $errorMsg error message text.
+     *
+     * @return nothing.
+     */
+    protected function addError(string $key, string $errorMsg)
+    {
+        $this->_errors[$key] = $errorMsg;
+    }
+
+    /**
+     * Accessor for field $_errors. The keys are the names of the LorisForm
+     * fields used for file uploads. Only fields with errors are stored in this
+     * array.
+     *
+     * @return array array of all the errors found.
+     */
+    public function getErrors() : array
+    {
+        return $this->_errors;
+    }
+
+    /**
+     * Accessor for field $_lorisForm.
+     *
+     * @return LorisForm the LorisForm associated with this handler.
+     */
+    public function getLorisForm() : \LorisForm
+    {
+        return $this->_lorisForm;
+    }
+
+    /**
+     * Accessor for field $_dirPermissions.
+     *
+     * @return int value for field $_dirPermissions.
+     */
+    public function getDirPermissions() : int
+    {
+        return $this->_dirPermissions;
+    }
+
+    /**
+     * Accessor for field $_moveDirPath.
+     *
+     * @return string value for field $_moveDirPath.
+     */
+    public function getMoveDirPath() : string
+    {
+        return $this->_moveDirPath;
+    }
+
+    /**
+     * Accessor for field $_dirGroup.
+     *
+     * @return int value for field $_dirGroup.
+     */
+    public function getDirGroup() : string
+    {
+        return $this->_dirGroup;
+    }
+
+    /**
+     * Accessor for field $_movedFiles.
+     *
+     * @return array the final location of all files uploaded on the
+     *               server. Each key is the name of a LorisFirm field
+     *               used to upload a file and its corresponding value is the final
+     *               location for the uploaded file on the server.
+     */
+    public function getMovedFiles() : array
+    {
+        return $this->_movedFiles;
+    }
+
+    /**
+     * Accessor for field $_dirPermissions.
+     *
+     * @return int value for field $_dirPermissions.
+     */
+    public function getFiles() : array
+    {
+        if (is_null($this->_files)) {
+            $this->_files = array();
+            foreach ($this->_lorisForm->getElements() as $element) {
+                if ($element instanceof LorisFormFileElement) {
+                    $this->_files[$element->getName()] = $element->getValue();
+                }
+            }
+        }
+
+        return $this->_files;
+    }
```

Comment:
This is unnecessary if you base it on ServerRequestInterface instead of LorisForm.